failure in tcp/test_ucp_sockaddr.concurrent_disconnect/7 on swx-rdmz-ucx-new-02

### Describe the bug
Happened here: https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/30728/logs/409

```
[ RUN      ] tcp/test_ucp_sockaddr.concurrent_disconnect/7 <tcp/stream,mt>
[     INFO ] Testing 127.0.0.1:0
[     INFO ] server listening on 127.0.0.1:56654
[swx-rdmz-ucx-new-02:1276 :4:1276] ucp_request.c:649  Assertion `(req->send.wireup.type == UCP_WIREUP_MSG_EP_REMOVED) || (req->send.wireup.type == UCP_WIREUP_MSG_EP_CHECK)' failed: req 0x9f51a20 ep 0x7f23b345b000: got ACK message

/scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_request.c: [ ucp_request_send_state_ff() ]
      ...
      643          * returned from 'uct_ep_pending_purge()', since they are released by
      644          * WIREUP endpoint's purge function
      645          */
==>   646         ucs_assertv((req->send.wireup.type == UCP_WIREUP_MSG_EP_REMOVED) ||
      647                     (req->send.wireup.type == UCP_WIREUP_MSG_EP_CHECK),
      648                     "req %p ep %p: got %s message", req, req->send.ep,
      649                     ucp_wireup_msg_str(req->send.wireup.type));

==== backtrace (tid:   1276) ====
 0 0x0000000000048374 ucp_request_send_state_ff()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_request.c:646
 1 0x0000000000048374 ucp_request_complete_send()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_request.inl:234
 2 0x0000000000048374 ucp_request_send_state_ff()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_request.c:676
 3 0x000000000002a2ee uct_tcp_ep_pending_purge_cb()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/uct/tcp/tcp_ep.c:2072
 4 0x0000000000056aeb uct_ep_pending_purge()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/uct/api/uct.h:3012
 5 0x0000000000037e7b ucp_ep_discard_lanes()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_ep.c:1446
 6 0x000000000003f9cc ucp_ep_set_failed()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_ep.c:1195
 7 0x00000000000eb881 ucp_ep_cm_remote_disconnect_progress()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/wireup/wireup_cm.c:847
 8 0x00000000000eb881 ucp_ep_cm_disconnect_progress()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/wireup/wireup_cm.c:873
 9 0x0000000000055c8b ucs_callbackq_slow_proxy()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucs/datastruct/callbackq.c:402
10 0x00000000000544d2 ucs_callbackq_dispatch()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucs/datastruct/callbackq.h:211
11 0x00000000000544d2 uct_worker_progress()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/uct/api/uct.h:2589
12 0x00000000000544d2 ucp_worker_progress()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../src/ucp/core/ucp_worker.c:2625
13 0x00000000009f873e ucp_test_base::entity::progress()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/ucp_test.cc:970
14 0x00000000009f873e ucp_test::progress()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/ucp_test.cc:168
15 0x00000000009d7de9 test_ucp_sockaddr::concurrent_disconnect()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/test_ucp_sockaddr.cc:578
16 0x00000000009984d3 test_ucp_sockaddr_concurrent_disconnect_Test::test_body()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/test_ucp_sockaddr.cc:790
17 0x00000000005ee296 ucs::test_base::run()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/test.cc:356
18 0x00000000005ee296 ucs::test_base::TestBodyProxy()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/test.cc:382
19 0x0000000000c80df9 HandleSehExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.cc:2433
20 0x0000000000c77c29 testing::Test::Run()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.cc:2509
21 0x0000000000c77d51 testing::TestInfo::Run()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.cc:2687
22 0x0000000000c77e15 testing::TestSuite::Run()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.cc:2819
23 0x0000000000c7893a testing::internal::UnitTestImpl::RunAllTests()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.cc:5350
24 0x0000000000c78af1 HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.cc:2433
25 0x00000000005764d5 RUN_ALL_TESTS()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/googletest/gtest.h:2473
26 0x00000000005764d5 main()  /scrap/azure/agent-08/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/main.cc:118
27 0x00000000000223d5 __libc_start_main()  ???:0
28 0x00000000005d1744 _start()  ???:0
=================================
[swx-rdmz-ucx-new-02:1276 :4:1276] Process frozen...
make: *** [test] Terminated

```

--- a/src/ucp/core/ucp_request.c
+++ b/src/ucp/core/ucp_request.c
@@ -638,13 +638,14 @@ void ucp_request_send_state_ff(ucp_request_t *req, ucs_status_t status)
     if (req->send.uct.func == ucp_proto_progress_am_single) {
         req->send.proto.comp_cb(req);
     } else if (req->send.uct.func == ucp_wireup_msg_progress) {
-        /* Sending EP_REMOVED/EP_CHECK WIREUP_MSGs could be scheduled on UCT
-         * endpoint which is not a WIREUP_EP. Other WIREUP MSGs should not be
-         * returned from 'uct_ep_pending_purge()', since they are released by
-         * WIREUP endpoint's purge function
+        /* Sending EP_REMOVED/EP_CHECK/ACK WIREUP_MSGs could be scheduled on
+         * UCT endpoint which is not a WIREUP_EP. Other WIREUP MSGs should not
+         * be returned from 'uct_ep_pending_purge()', since they are released
+         * by WIREUP endpoint's purge function
          */
         ucs_assertv((req->send.wireup.type == UCP_WIREUP_MSG_EP_REMOVED) ||
-                    (req->send.wireup.type == UCP_WIREUP_MSG_EP_CHECK),
+                    (req->send.wireup.type == UCP_WIREUP_MSG_EP_CHECK) ||
+                    (req->send.wireup.type == UCP_WIREUP_MSG_ACK),
                     "req %p ep %p: got %s message", req, req->send.ep,
                     ucp_wireup_msg_str(req->send.wireup.type));
         ucs_free(req->send.buffer);

